### ISSUE TITLE
Cache moroutines?

### ISSUE THREAD

**Issue by YeudielG** [KurbanismailovZaur/Moroutines#1]
Hi!

Is it possible to cache the moroutines? Something like:
```
Moroutine mor;

void Start() 
{
      mor = Moroutine.Create(CoroutineName());
}

void DoSomething() 
{
      mor.Run();
}
```

I'd like to cache them, given the vast amount of coroutines I have, though I'm not sure if it's necessary with this ones. Still, is it possible? I was trying to but the console gave me this error: `PlayControlException: Moroutine already destroyed.` Which I'm not sure if it's expected or not.

**Comment by KurbanismailovZaur** [comment]
Hi bro, of course you can cache moroutines. But you need to know about how
IEnumerator methods works.. If you use yield return in IEnumerator, then
generated enumerator object can't be reseted after finishing playing. It is
because generated IEnumerator objects don't implements Reset method. So if
you pass enumerator object in moroutines Create method they can play
enumerator only once and after playing is finished moroutine become
destroyed. You can't use destroyed moroutines ever. That why you take
message about PlayControlException. Still, you can change this behaviour
with mor.SetAutoDestroy(false) method, but this doesn't make sense, because
enumerator object won't do anything.

Hoverer, you can generate IEnumerable object instead IEnumerator. This
gives you apportunity to recreate IEnumerators with GetEnumerator method.
So if you pass enumerable to moroutine it can repeat playing as much as you
want. Autodestroying by default is set to false in that moroutines. And
here it makes sense to use caching.

I was update moroutines in last few days. You can download last version at:
https://github.com/KurbanismailovZaur/Moroutines/releases

And also I update docs. Hope I was help you, bro!

With love, from Russia! ❤️

Вс, 10 апр. 2022 г. в 11:04, theMasky ***@***.***>:

**Comment by KurbanismailovZaur** [comment]
Did I help you?

Вс, 10 апр. 2022 г. в 13:45, Zaur Kurbanismailov <
***@***.***>:

**Comment by YeudielG** [comment]
@KurbanismailovZaur Thanks! Yes, that was really helpful.

**Comment by KurbanismailovZaur** [comment]
Nice! If you like my asset, please rate it on asset store: Moroutines
<https://assetstore.unity.com/packages/tools/utilities/moroutines-214820#reviews>
Maybe the following assets will also interest you:

   - Await Extensions
   <https://assetstore.unity.com/packages/tools/integration/await-extensions-127006#description>
-
   async/await support in Unity.
   - Extensions
   <https://assetstore.unity.com/packages/tools/utilities/extensions-214756#publisher>
-
   too many methods for everyday programming in Unity.

Best Regards!

пн, 11 апр. 2022 г. в 00:24, theMasky ***@***.***>:
